### PR TITLE
Fix clippy warnings

### PR DESCRIPTION
**Pull request by sushi-shi**
It might not be necessary, but I still fixed all clippy warnings. It might also be reasonable to add clippy stage into CI, so that no warnings are introduced further on

### CODE REVIEW

**Comment by sushi-shi** [comment]
This small script changed quite a few files. I wonder whether this is okay or it might be better to change them incrementally.
```bash
[sheep]$ set files (rg 'JoshutoResult<\(\)>' --files-with-matches)
[sheep]$ for file in $files; sd -s 'JoshutoResult<()>' 'JoshutoResult' $file ; end
```

File: src/context/worker_context.rs [hunk 1]
```diff
@@ -29,7 +29,8 @@ impl WorkerContext {
     // worker related
     pub fn push(&mut self, thread: IoWorkerThread) {
         self.worker_queue.push_back(thread);
-        self.event_tx.send(AppEvent::IoWorkerCreate);
+        // error is ignored
+        let _ = self.event_tx.send(AppEvent::IoWorkerCreate);
```
Comment (sushi-shi) [inline src/context/worker_context.rs:33]: I also wonder whether here should be `unwrap`.

Comment (kamiyaa) [reply src/context/worker_context.rs:33]: Should be fine

**Review by kamiyaa — APPROVED** [review]

**Comment by kamiyaa** [comment]
Thanks for the work!

**Pull request merged**